packages/addon/src/utils.ts

```diff
@@ -1,11 +1,16 @@
 import { Cookie } from './types';
 
-export function setCookie(name: string, value: unknown) {
-  document.cookie = `${name}=${JSON.stringify(value)}`;
+export function setCookie(name: string, value: string) {
+  if (typeof value !== 'string') {
+    document.cookie = `${name}=${JSON.stringify(value)};`;
+  } else {
+    const encodedValue = encodeURIComponent(value);
```

Comment:
encode cookies to follow specifications for cookie-octet  https://datatracker.ietf.org/doc/html/rfc6265#section-4.1.1
```

%x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
; US-ASCII characters excluding CTLs,
; whitespace DQUOTE, comma, semicolon,
; and backslash
```